Sources/SubscriberAttributes/CustomAttributesManager.swift

```diff
@@ -0,0 +1,161 @@
+            Logger.warn(
+                Strings.identity.sync_attributes_and_offerings_rate_limit_reached(
+                    maxCalls: manualRateLimiter.maxCalls,
+                    period: Int(manualRateLimiter.period)
+                )
+            )
+            self.getOfferings(appUserID: appUserID,
+                              fetchPolicy: .default,
+                              fetchBehavior: .cachedOrFetched,
+                              completion: completion)
+            return
+        }
+
+        self.syncSubscriberAttributes(
+            appUserID: appUserID,
+            attribution: attribution,
+            completion: {
+                self.getOfferings(appUserID: appUserID,
+                                  fetchPolicy: .default,
+                                  fetchBehavior: .fetchCurrent(reason: .manualSyncCustomAttributes),
+                                  completion: completion)
+        })
+
+    }
+
+    func syncCustomAttributesAndOfferingsIfNeeded(
+        appUserID: String,
+        attribution: Attribution,
+        subscriberAttributionsManager: SubscriberAttributesManager
+    ) {
+        let hasUnsyncedCustomAttributes = self.hasUnsyncedCustomAttributes(
+            appUserID: appUserID,
+            attribution: attribution,
+            subscriberAttributionsManager: subscriberAttributionsManager
+        )
+
+        if hasUnsyncedCustomAttributes {
+            guard automaticRateLimiter.shouldProceed() else {
+                Logger.warn(
+                    Strings.identity.sync_custom_attributes_rate_limit_reached(
+                        maxCalls: manualRateLimiter.maxCalls,
+                        period: Int(manualRateLimiter.period)
+                    )
+                )
+                return
+            }
+
+            automaticThrottler.throttle {
+                self.syncSubscriberAttributes(
+                    appUserID: appUserID,
+                    attribution: attribution,
+                    completion: {
+                        self.getOfferings(appUserID: appUserID,
+                                          fetchPolicy: .default,
+                                          fetchBehavior: .fetchCurrent(reason: .automaticSyncCustomAttributes),
+                                          completion: { _, _ in })
+                })
+            }
+        }
+    }
+
+}
+
+private class Throttler {
+    private var workItem: DispatchWorkItem?
+    private let queue: DispatchQueue
+    private let delayInSeconds: Double
+
+    init(delayInSeconds: Double, queue: DispatchQueue = DispatchQueue.main) {
+        self.delayInSeconds = delayInSeconds
+        self.queue = queue
+    }
+
+    func throttle(_ block: @escaping () -> Void) {
+        workItem?.cancel()
+
+        let task = DispatchWorkItem { block() }
+        self.workItem = task
+
+        queue.asyncAfter(deadline: .now() + delayInSeconds, execute: task)
+    }
+}
```

Comment:
This could be moved into its own file where `RateLimiter` is. Not sure if this is the best approach for this (or if we have something existing that I could use) but it wanted to keep it simple at first